### PR TITLE
Update to libdjinterop 0.19

### PR DESCRIPTION
**Pull request by mr-smidge**
This PR updates Mixxx to use version 0.19 of libdjinterop, which supports Denon firmware up to version 3.0.0.

There are no functional changes as part of this PR - the changes are limited to those necessary to adhere to the evolving public API of libdjinterop whilst it is in its early 0.x state.

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
The `HotcueControlTest.SavedLoopMove` fail in `build-checks / coverage` is unrelated!

**Comment by JoergAtGithub** [comment]
@toszlanyi @whanake-music Could you please check that this PR works with your EngineOS devices.

File: src/library/export/dlglibraryexport.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@
 #include "library/trackset/crate/cratestorage.h"
 #include "moc_dlglibraryexport.cpp"
 
-namespace el = djinterop::enginelibrary;
+namespace el = djinterop::engine;
```
Comment (JoergAtGithub) [inline src/library/export/dlglibraryexport.cpp:21]: el isn't self explaining anymore, if you remove the word library

**Comment by toszlanyi** [comment]
> @toszlanyi @whanake-music Could you please check that this PR works with your EngineOS devices.

I successfully built the PR this morning and gonna check exporting and playing for the Twitch stream tonight. Report back tomorrow. Cheers!

**Comment by toszlanyi** [comment]
> >... Report back tomorrow. Cheers!
> 

All went smooth on my SC6000M, except one track (out of ca. 150) that wouldn't show the overview waveform on the player. It's available in Mixxx though. I gonna do more testing and discuss in zulip chat for more details. All in all a nice experience for my 2h set yesterday. Cheers

**Comment by JoergAtGithub** [comment]
@toszlanyi Is the waveform issue new, or did it occur already with djinterop 0.16.1 which is currently included in Mixxx?

File: src/library/export/dlglibraryexport.cpp (hunk 1 above)
Comment (mr-smidge) [reply src/library/export/dlglibraryexport.cpp]: Good spot - renamed to `e`.

**Comment by toszlanyi** [comment]
> @toszlanyi Is the waveform issue new, or did it occur already with djinterop 0.16.1 which is currently included in Mixxx?

I did test a couple of scenarios and unfortunately need to admit it is a new issue. I gonna give a full report on [zulip chat](https://mixxx.zulipchat.com/#narrow/stream/109171-development/topic/Export.20to.20Engine.20Prime)

File: src/library/export/dlglibraryexport.cpp [hunk 2]
```diff
@@ -222,56 +222,42 @@ void DlgLibraryExport::exportRequested() {
...
+        bool exists = e::database_exists(databaseDirectory.toStdString());
         if (!exists) {
             // The user can freely choose a schema version for their new database.
             m_pExistingDatabaseLabel->setText("");
             m_pVersionCombo->clear();
             m_pVersionCombo->setEnabled(true);
-            int versionIndex = 0;
-            for (const djinterop::semantic_version& version : el::all_versions) {
+            for (int versionIndex = 0;
+                    versionIndex < (int)e::all_versions.size();
```
Comment (Swiftb0y) [inline src/library/export/dlglibraryexport.cpp:236]:
```suggestion
            for (int versionIndex = 0;
                    versionIndex < static_cast<int>(e::all_versions.size());
```

**Review by Swiftb0y — COMMENTED** [review]
rest lgtm.

File: src/library/export/dlglibraryexport.cpp [hunk 3]
```diff
@@ -233,7 +233,7 @@ void DlgLibraryExport::checkExistingDatabase() {
             m_pVersionCombo->clear();
             m_pVersionCombo->setEnabled(true);
             for (int versionIndex = 0;
-                    versionIndex < (int)e::all_versions.size();
+                    versionIndex < static_cast<int>(e::all_versions.size());
```
Comment (daschuer) [inline src/library/export/dlglibraryexport.cpp:236]: Why not use std::size_t in the first place and remove the nerrowing cast here?

Comment (Swiftb0y) [reply src/library/export/dlglibraryexport.cpp:236]: because we add `-1` further down.

**Comment by JoergAtGithub** [comment]
@mr-smidge We have merge conflicts in CMakeLists.txt now - due to merge of #11729 into 2.4. Please resolve the conflicts.

**Comment by JoergAtGithub** [comment]
@mr-smidge Should we merge this PR as it is, to have a bigger testing group - or do you plan fixes for the missing waveform issue reported above first?

**Comment by mr-smidge** [comment]
> @mr-smidge Should we merge this PR as it is, to have a bigger testing group - or do you plan fixes for the missing waveform issue reported above first?

Realistically, at this stage I don't know what form the waveform fix will take - my debugging will essentially be to change things methodically until I find the thing that makes it work!

It seems from testing that the waveform issue only affects a minority of tracks at present, and there is still a workaround: to export as Engine version 1.6.1 (which does not display the bug) and let the Denon players upgrade the database to 3.x.

I would be comfortable merging this in its current state, but I appreciate the decision is not up to me :smile:.

**Comment by JoergAtGithub** [comment]
Ok, than let's merge it! Thank you for this big improvement, we know it was a huge amount of work!

**Pull request merged**